kubernetes/resource_kubernetes_mutating_webhook_configuration.go

```diff
@@ -0,0 +1,325 @@
+							Type:        schema.TypeList,
+							Description: webhookDoc["admissionReviewVersions"],
+							Optional:    true,
+							Elem:        &schema.Schema{Type: schema.TypeString},
+						},
+						"client_config": {
+							Type:        schema.TypeList,
+							Description: webhookDoc["clientConfig"],
+							Required:    true,
+							MaxItems:    1,
+							Elem: &schema.Resource{
+								Schema: webhookClientConfigFields(),
+							},
+						},
+						"failure_policy": {
+							Type:        schema.TypeString,
+							Description: webhookDoc["failurePolicy"],
+							Optional:    true,
+							Default:     "Fail",
+						},
+						"match_policy": {
+							Type:        schema.TypeString,
+							Description: webhookDoc["matchPolicy"],
+							Optional:    true,
+							Default:     "Equivalent",
+						},
+						"name": {
+							Type:        schema.TypeString,
+							Description: webhookDoc["name"],
+							Required:    true,
+						},
+						"namespace_selector": {
+							Type:        schema.TypeList,
+							Description: webhookDoc["namespaceSelector"],
+							Optional:    true,
+							MaxItems:    1,
+							Elem: &schema.Resource{
+								Schema: labelSelectorFields(true),
+							},
+						},
+						"object_selector": {
+							Type:        schema.TypeList,
+							Description: webhookDoc["objectSelector"],
+							Optional:    true,
+							MaxItems:    1,
+							Elem: &schema.Resource{
+								Schema: labelSelectorFields(true),
+							},
+						},
+						"reinvocation_policy": {
+							Type:        schema.TypeString,
+							Description: webhookDoc["reinvocationPolicy"],
+							Optional:    true,
+							Default:     "Never",
+						},
+						"rule": {
+							Type:        schema.TypeList,
+							Description: webhookDoc["rules"],
+							Required:    true,
+							MinItems:    1,
+							Elem: &schema.Resource{
+								Schema: ruleWithOperationsFields(),
+							},
+						},
+						"side_effects": {
+							Type:        schema.TypeString,
+							Description: webhookDoc["sideEffects"],
+							Optional:    true,
+						},
+						"timeout_seconds": {
+							Type:        schema.TypeInt,
+							Description: webhookDoc["timeoutSeconds"],
+							Default:     10,
+							Optional:    true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceKubernetesMutatingWebhookConfigurationCreate(d *schema.ResourceData, meta interface{}) error {
+	var err error
+
+	conn, err := meta.(KubeClientsets).MainClientset()
+	if err != nil {
+		return err
+	}
+
+	cfg := admissionregistrationv1.MutatingWebhookConfiguration{
+		ObjectMeta: expandMetadata(d.Get("metadata").([]interface{})),
+		Webhooks:   expandMutatingWebhooks(d.Get("webhook").([]interface{})),
+	}
+
+	log.Printf("[INFO] Creating new MutatingWebhookConfiguration: %#v", cfg)
+
+	res := &admissionregistrationv1.MutatingWebhookConfiguration{}
+
+	usev1beta1, err := useAdmissionregistrationV1beta1(conn)
+	if err != nil {
+		return err
+	}
+	if usev1beta1 {
```

Comment:
would it make more sense if we called it something more explicit, like `useadmissionregistrationv1beta1` ?

Reply:
No, that's ok. It was the request/response stuff that I was having trouble reading. `resv1beta1` and `reqv1beta1`. Maybe if they were named `request` and `response` instead, it would be easier to read. But it's ok if you don't want to change it too.

Reply:
Ah gotcha, I'm good with changing the names to be more explicit.